Add lua memory option (#4736)
tarantool/doc: doc/reference/configuration/configuration_reference.rst

    | Default: box.NULL
    | Environment variable: TT_LOG_SYSLOG_SERVER

.. _configuration_reference_lua:

lua
---

> The ``lua`` section outlines the configuration parameters related to the ``lua`` environment within Tarantool.

Reviewer: "Lua" is not highlighted as code when it's a proper name (not a keyword). So "``lua`` environment" is just "Lua environment". 

Same for the ``lua.memory`` option description.

> Below is an overview of the available configuration options for ``lua``:

Reviewer: We keep introductory texts in big references to the minimum, so this sentence looks redundant. This is also consistent with other sections of this page.

- :ref:`lua.memory <configuration_reference_lua>`

> 

Reviewer: Please check the [configuration scope](https://www.tarantool.io/en/doc/latest/platform/configuration/#configuration-scopes) to which the ``lua`` section applies.
I'll explain more in a call.

.. _configuration_reference_lua_memory:

.. confval:: lua.memory

Specifies the amount of memory allocated to ``lua``, measured in bytes.

> - **Type**: integer

Reviewer: Please use consistent markup as in other option descriptions.

> - **Default Value**: 2GB

Reviewer: Since it's a number, lets also show the exact value in bytes. Find similar examples in other config options.

- **Minimum Value**: 256MB

> 

Reviewer: Add linked environment variable name.
More details in a call ;)

> **Dynamic adjustment behavior:**

Reviewer: Pls don't use bold for "like-a-heading" highlighting.
I'd move this info to the option description and use the regular text style for consistency with other options.

    | Default: box.NULL
    | Environment variable: TT_LOG_SYSLOG_SERVER

.. _configuration_reference_lua:

lua
---

The ``lua`` section outlines the configuration parameters related to the Lua environment within Tarantool.

.. NOTE::

    ``lua`` can be defined in any :ref:`scope <configuration_scopes>`.

> ..  NOTE::

Reviewer: Pls remove this line.

- :ref:`lua.memory <configuration_reference_lua>`

.. _configuration_reference_lua_memory:

.. confval:: lua.memory

> Specifies the amount of memory allocated to lua, measured in bytes.

Reviewer 2: Nit: This could be misleading since `allocated` means that the specified amount of memory will for sure be consumed. Whereas the parameter only sets the limit (and so it can be greater than the memory available to the machine).

And there are variations in how `in bytes` is written as measurement units in the docs. Other options express it either like `option specifies X (in bytes)` or like `option specifies X in bytes`. Though `measured in bytes` is a new one. Let's tend to one of the used for consistency.

```suggestion
Specifies the maximum amount of memory available to Lua scripts (in bytes).
```

>  |

Reviewer: Indent looks incorrect compared to other options, please check and fix:
<img width="650" alt="image" src="https://github.com/user-attachments/assets/de60ff16-fa98-45ab-b54c-b4baab7af4be" />

 | Type: integer

>  | Default Value: 2147483648 (2GB)

Reviewer: ```suggestion
 | Default: 2147483648 (2GB)
```
For consistency with other options

>  | Environment variable: TT_LUA_MEMORY for the cast value and TT_LUA_MEMORY_DEFAULT for the default value

Reviewer 2: Nit: AFAIS all other parameter descriptions don't provide the information on the `DEFAULT` environment variable, since all the variable share the same pattern `TT_<VARNAME>` for the value and `TT_<VARNAME>_DEFAULT` for the default value.

```suggestion
 | Environment variable: TT_LUA_MEMORY
```

When the specified value exceeds the current memory usage, the new limit takes effect immediately without a restart.
However, **when the specified value is lower than the current memory usage, a restart of the instance is required** for the change to take effect.

> Example to set the Lua memory limit for 4 GB:

Reviewer 2: Nit: I am almost sure that `to` should be used instead of `for`.

```suggestion
Example of setting the Lua memory limit to 4 GB:
```

Another option is to follow common `The example below shows...` pattern.

```suggestion
The example below shows how to limit the memory used by Lua to 4 GB:
```

Reviewer 2: @maryiaLichko what about this? Am I wrong that `to` should be used here instead of `for`? I'd wish some clarification here.

Author: @georgiy-belyanin the sentence is grammatically correct. Both variants are ok

Reviewer 2: But it seems `for` is used for specifying the target for which the limit is applied.
I refer to these two links.
https://context.reverso.net/translation/english-russian/set+limit+for
https://context.reverso.net/translation/english-russian/set+limit+to

I am ok if you provide a source where `for` is used with this meaning.

Reviewer 2: I see you applied. Resolving then. Thanks.

Author: changed to `to` because` for` is used for the purpose or scope of the limit, not the specific value :)
thanks
